### ISSUE TITLE
Classifier zoom functions called twice on each click of Zoom In or Zoom Out

### ISSUE THREAD

**Issue by goplayoutside3** [zooniverse/front-end-monorepo#5043]
## Package
lib-classifier

## Describe the bug

When I click the Zoom In Button or Zoom Out Button in the Image Toolbar, the zoom-in and zoom-out functions are called twice. I can see the subject image zoom twice and a message of "zooming in" or "zooming out" is printed twice in the browser console.

## To Reproduce

Go to any project's classify page that uses images or json subject (i.e [TESS](https://www.zooniverse.org/projects/nora-dot-eisner/planet-hunters-tess/classify)) and click on the zoom buttons in the image toolbar. The subject will zoom twice instead of once, as expected

**Comment by eatyourgreens** [comment]
I think this event handler is missing `event.preventDefault()` (to cancel the click for pointers.)

https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/ImageToolbar/components/ZoomInButton/ZoomInButtonContainer.js#L28-L35

**Comment by eatyourgreens** [comment]
The click event is working as expected when it's fired by the Enter key on either Zoom In, or Zoom out. The callback is only called once for each click.

When I click or press with a pointer, I'm seeing `pointerdown` while the pointer is down,  then `click` and `pointerup` when I release the pointer, which calls `zoomIn()` (or `zoomOut()`) at least twice.

**Comment by eatyourgreens** [comment]
Other odd behaviour: when I hold Enter down, the button fires continuously. That’s useful here, but I’d expect `click` to fire once, when the button is released.

**Comment by eatyourgreens** [comment]
Removing the `onClick` handler from these buttons will fix this bug.

**Comment by eatyourgreens** [comment]
This seems to fix the bug, plus gets the separate frames view working when you hold the zoom buttons down. Ignore the click event if there's already a timer, set by a pointer:

```js
const zoomCallback = separateFramesView ? separateFrameZoomIn : zoomIn

  function onClick() {
    if (!timer) {
      zoomCallback()
    } else {
      clearInterval(timer)
      setTimer('')
    }
  }

  function onPointerDown(event) {
    const { currentTarget, pointerId } = event
    zoomCallback()
    clearInterval(timer)
    const newTimer = setInterval(zoomCallback, 100)
    setTimer(newTimer)
    currentTarget.setPointerCapture(pointerId)
  }

  function onPointerUp(event) {
    const { currentTarget, pointerId } = event
    currentTarget.releasePointerCapture(pointerId)
  }
```

**Comment by eatyourgreens** [comment]
Also, I had the order of events wrong earlier. In Firefox, it's `pointerdown`, `pointerup` then `click`.

**Comment by eatyourgreens** [comment]
These tests don't check that each callback is only called once. I've fixed that in #5050.
https://github.com/zooniverse/front-end-monorepo/blob/09d4322ebe6aa36ba961fc4c01e3a68077b5cde5/packages/lib-classifier/src/components/Classifier/components/ImageToolbar/components/ZoomInButton/ZoomInButton.spec.js#L46-L61

**Comment by goplayoutside3** [comment]
Closed as of #5050
